src/DynamoCore/Utilities/Extensions.cs

```diff
@@ -29,6 +29,25 @@ public interface ITree<out TNodeTag, out TLeaf>
     public static class ExtensionMethods
     {
         /// <summary>
+        /// Get the index of an element in an IEnumerable
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="source">The IEnumerable</param>
+        /// <param name="value">The value for which the index is sought</param>
+        /// <returns>Zero or greater if in the IEnumerable, otherwise -1</returns>
+        public static int IndexOf<T>(this IEnumerable<T> source, T value)
```

Comment:
Where is this new extension method used? Did you make this then realize IEnumerable has ElementAt?